### PR TITLE
Throw 404 exception when site not found on routing fail

### PR DESCRIPTION
**Pull request by ngilain**

### CODE REVIEW

File: FrontBundle/EventSubscriber/KernelExceptionSubscriber.php [hunk 1]
```diff
@@ -94,10 +96,11 @@ protected function getCurrentSiteInfo($host, $path)
             }
         }
 
-        return array(
-            'site' => $possibleSite,
-            'language' => $possibleAlias->getLanguage()
-        );
+        if (is_null($possibleAlias)) {
+            throw new NonExistingSiteException();
```
Comment (alavieille) [inline FrontBundle/EventSubscriber/KernelExceptionSubscriber.php:100]: may be update the test for this case ?

**Comment by alavieille** [comment]
else test :+1:

**Comment by ngilain** [comment]
@itkg-alavieille : tests updated

File: FrontBundle/Tests/EventSubscriber/KernelExceptionSubscriberTest.php [hunk 2]
```diff
@@ -103,4 +103,35 @@ public function getErrorContext()
             array('500', $node, 0),
         );
     }
+
+    /**
+     * @param string $status
+     * @param int    $expectedResponseCount
```
Comment (alavieille) [inline FrontBundle/Tests/EventSubscriber/KernelExceptionSubscriberTest.php]: update php doc

**Comment by nicolasThal** [comment]
:+1:

**Comment by alavieille** [comment]
else phpdoc :+1:

**Pull request merged**